src/main/java/fr/polytech/sim/transport/Vehicle.java

```diff
@@ -1,18 +1,19 @@
 package fr.polytech.sim.transport;
 
-import fr.polytech.sim.log.ConsoleLogger;
 import fr.polytech.sim.log.Logger;
-import java.util.HashSet;
+import fr.polytech.sim.log.LoggerFactory;
+
+import java.util.ArrayList;
 import java.util.Iterator;
-import java.util.Set;
+import java.util.List;
 
 /**
  * Vehicle abstraction.
  */
 public abstract class Vehicle implements MobileObject {
 
-    private final Logger logger = new ConsoleLogger("Vehicle");
-    protected final Set<MobileObject> components = new HashSet<>();
+    private final Logger logger = LoggerFactory.createLogger("Vehicle");
+    protected final List<MobileObject> components = new ArrayList<MobileObject>();
```

Comment:
Ce n'est pas une erreur, mais on pourrait supprimer le paramètre de type ici (cf. diamond operator).
```suggestion
    protected final List<MobileObject> components = new ArrayList<>();
```